### ISSUE TITLE
KeyNotFoundException when building models

### ISSUE THREAD

**Issue by robertjf** [callumbwhyte/super-value-converters#5]
I've inherited a site that causes the following error when attempting to build models:

```
Failed to build models.
The given key was not present in the dictionary.

at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
at Our.Umbraco.SuperValueConverters.PreValues.PreValueMapper.Map[T](T model, IDictionary`2 preValues)
at Our.Umbraco.SuperValueConverters.ValueConverters.MediaPickerValueConverter.GetSettings(PublishedPropertyType propertyType)
at Our.Umbraco.SuperValueConverters.ValueConverters.MediaPickerValueConverter.GetPropertyValueType(PublishedPropertyType propertyType)
at Umbraco.Core.Models.PublishedContent.PublishedPropertyType.InitializeConverters()
at Umbraco.Core.Models.PublishedContent.PublishedPropertyType..ctor(PublishedContentType contentType, PropertyType propertyType)
at Umbraco.Core.Models.PublishedContent.PublishedContentType.<.ctor>b__3_0(PropertyType x)
at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
```

The problem shows up with the MediaPicker built-in datatype - for some reason not all the Prevalues are populated until I go to it in DataTypes and save it.

Adding a check for the prevalue property alias in the prevalues collection (in `PreValueMapper`) resolves this.  (I'll do a pull request for you)

**Comment by callumbwhyte** [comment]
Hey @robertjf!

Thanks for raising this - and thanks for trying out SuperValueConverters! This is an interesting issue...

I’m going to do some investigation as to why this happens in the first place... Could you specify which Umbraco version this was tested on?

Callum

**Comment by callumbwhyte** [comment]
This is fixed PR #6 and due to be released in v1.0.1

**Comment by robertjf** [comment]
@callumbwhyte it was Umbraco 7.13.1 which I had also upgraded to 7.13.2 - I have noticed this behaviour before (missing prevalue data) when working with uSync before but can't definitively say that is the cause...

**Comment by callumbwhyte** [comment]
@robertjf Your fix has been released in v1.0.1, available from all good package stores (Our Umbraco, NuGet, Github).

Thanks for the hard work! 👍
